File: examples/java/taxi/src/main/java/be/kuleuven/taxi/Driver.java
package be.kuleuven.taxi;

public class Driver {
    private final String name;
    private final int skillLevel;
    private boolean working = false;

    public String getName() {
        return name;
    }

    public int getSkillLevel() {
        return skillLevel;
    }

    public Driver drive(Client client) {
        working = true;
        return this;
    }

    public boolean isWorking() {
        return working;
    }

    public void with(Car car) {
        if(car.isDriving()) {
            throw new CarIsAlreadyDrivenExceptino("Sorry boss, I can't, my colleague is already driving this car!");
        }
        car.startEngine();
    }

    public Driver(String name, int skillLevel) {
        this.name = name;
        this.skillLevel = skillLevel;
    }
}

File: examples/kotlin/taxi/src/main/kotlin/be/kuleuven/taxis/Driver.kt
package be.kuleuven.taxis

class Driver(name: String, skillLevel: Int) {
    val name: String = name
    val skill: Int = skillLevel

    fun drive(client: Client): Driver {
        working = true
        return this
    }

    // it's NOT "Car car" but "car: Car". In kotlin, it's FIRST the name, THEN the type.
    fun with(car: Car) {
        if(car.driving) {
            throw CarIsAlreadyDrivenException("Sorry boss, I can't, my colleague is already driving this car!")
        }
        car.startEngine()
    }

    var working: Boolean = false
}